### ISSUE TITLE
Update `currentBrowser` Core util and browser naming conventions

### ISSUE THREAD

**Issue by haydnba** [exteranto/framework#147]
`currentBrowser` method of `core/src/support/Utils` needs to be updated - currently, seems not to properly handle Safari browser feature detection adequately. 

Minimally we should safely handle Chrome, Firefox, Safari, Opera, MS Edge.

Also, we should update the browser name conventions found in `core/src/support/Browser` enum with 'FIREFOX' instead of 'EXTENSIONS' in preparation for API wrapper updates....

**Comment by bausano** [comment]
The original motivation behind naming it EXTENSIONS was that FF adheres to a standard which seemed to be shared also with Microsoft Edge and Opera.

**Comment by haydnba** [comment]
Yes, it seems now that Safari will reintroduce JavaScript based browser extensions that they are following closely the 'browser' namespace WebExtensions API used by Firefox (although it is equally possible to build the Safari extension using 'chrome' namespace as well, since they have created their own build tool). Meanwhile, Chrome, Edge (MS) and Opera are all using the 'chrome' namespace. So we have a situation now that half browsers use 'browser' and other half use 'chrome'....
